.github/workflows/deployment.yml

```diff
@@ -0,0 +1,171 @@
+name: Publish to PyPI
+
+on:
+  push:
+    paths:
+      - src/mrpro/VERSION
+
+jobs:
+  build-testpypi-package:
+    name: Modify version and build package for TestPyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: Set environment variable for version suffix
+        run: echo "MRPROVERSIONSUFFIX=.pre$(date +%s)" >> $GITHUB_ENV
+
+      - name: Build package with version suffix
+        run: |
+          python -m pip install --upgrade build
+          python -m build
+
+      - name: Store TestPyPi distribution
+        uses: actions/upload-artifact@v4
+        with:
+          name: testpypi-package-distribution
+          path: dist/
+
+  testpypi-deployment:
+    name: Deploy to TestPyPI
+    needs:
+      - build-testpypi-package
+    runs-on: ubuntu-latest
+
+    environment:
+      name: testpypi
+      url: https://test.pypi.org/p/mrpro
+
+    permissions:
+      id-token: write
+
+    steps:
+      - name: Download TestPyPi distribution
+        uses: actions/download-artifact@v4
+        with:
+          name: testpypi-package-distribution
+          path: dist/
+
+      - name: Publish to TestPyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
+          verbose: true
+
+  test-install-from-testpypi:
+    name: Test installation from TestPyPI
+    needs:
+      - testpypi-deployment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+
+      - name: Install MRpro from TestPyPI
+        run: |
+          python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ mrpro
```

Comment:
i am not sure this does what we think it does.
wont this default to installing the last released version?
at the minimum, we might need a --pre here to allow rc versions.

Better would be to actually try to install the version we just uploaded.

Reply:
should be the case now... I store the version and suffix variables in the `build-testpypi-package` job now:

```
VERSION=$(cat src/mrpro/VERSION)
SUFFIX=rc$(date +%s)
echo "suffix=$SUFFIX" >> $GITHUB_OUTPUT
echo "version=$VERSION" >> $GITHUB_OUTPUT
```
and then use it later in the `test-install-from-testpypi` job:

```
VERSION=${{ needs.build-testpypi-package.outputs.version }}
SUFFIX=${{ needs.build-testpypi-package.outputs.suffix }}
python -m pip install mrpro==$VERSION$SUFFIX --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/
```